src/query-builder/SelectQueryBuilder.ts

```diff
@@ -2051,6 +2071,7 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
                 type: "join",
                 name: aliasName,
                 metadata: joinAttributeMetadata,
+                subQuery: isSubQuery ? subQuery : undefined,
```

Comment:
When subquery is provided, https://github.com/typeorm/typeorm/blob/58c243d07c319a4eb4b218889665de0d04c8f4ff/src/query-builder/SelectQueryBuilder.ts#L2276-L2278 `createJoinExpression` already prefers using a subQuery if present, it doesn't seem to have any other effect besides this.

If there is any potential problem of passing `subQuery` here, please chime in.